infra/dockerfile.commit_retrieval

```diff
@@ -2,18 +2,22 @@ FROM mozilla/bugbug-base:latest
 
 COPY infra/git_hyper_blame_patch /root/git_hyper_blame_patch
 
-# Mercurial need Python2 :(
+# Setup mercurial from its own website, without install pip2 which has a lot of dependencies
+ARG MERCURIAL_VERSION="5.1"
+
 # git and git-hyper-blame (from depot_tools) are required by the annotate pipeline.
 RUN apt-get update && \
-    apt-get install -y python python-pip git curl && \
-    python2 -m pip install --disable-pip-version-check --no-cache-dir mercurial==5.1 && \
+    apt-get install --no-install-recommends -y python python-dev python-docutils build-essential git curl && \
+    curl -L https://www.mercurial-scm.org/release/mercurial-$MERCURIAL_VERSION.tar.gz | tar -C /opt -xvz && \
+    make -C /opt/mercurial-$MERCURIAL_VERSION/ install PYTHON=/usr/bin/python2 && \
+    hg debuginstall && \
```

Comment:
Why are you installing in this way instead of just copying from https://github.com/mozilla/code-coverage/blob/master/bot/ci/bootstrap.sh? :P

Reply:
I was using the python install method earlier but It was giving a mysterious error which I finally managed to solve. So I'll be changing this in the next edit.

Reply:
I haven't come across the method of installing mercurial 5.1 using python other than here (https://www.mercurial-scm.org/wiki/UnixInstall).
The official documentation only had the instructions to install it via make, https://www.mercurial-scm.org/wiki/UnixInstall